### PR TITLE
[Project] Add delete artifact method

### PR DESCRIPTION
**Pull request by moranbental**
Creating a new method in MlrunProject that allows the user to delete an artifact object in the DB and optionally delete the artifact data.

```
artifact = project.get_artifact("test-delete")
project.delete_artifact(artifact, deletion_strategy=mlrun.common.schemas.artifact.ArtifactsDeletionStrategies.data_optional, secrets={"secret1": "user-secret"})
```

https://iguazio.atlassian.net/browse/ML-6295

### CODE REVIEW

**Review by liranbg — APPROVED** [review]

File: mlrun/db/httpdb.py [hunk 1]
```diff
@@ -985,14 +985,27 @@ def read_artifact(
...
+        secrets: dict = None,
+    ):
         """Delete an artifact.
 
         :param key: Identifying key of the artifact.
         :param tag: Tag of the artifact.
         :param project: Project that the artifact belongs to.
         :param tree: The tree which generated this artifact.
         :param uid: A unique ID for this specific version of the artifact (the uid that was generated in the backend)
+        :param deletion_strategy: The artifact deletion strategy types.
```
Comment (alonmr) [inline mlrun/db/httpdb.py:1007]: point to where they can see the types

File: mlrun/projects/project.py [hunk 2]
```diff
@@ -1593,6 +1594,23 @@ def log_artifact(
...
+        item: Artifact,
+        deletion_strategy: mlrun.common.schemas.artifact.ArtifactsDeletionStrategies = (
+            mlrun.common.schemas.artifact.ArtifactsDeletionStrategies.metadata_only
+        ),
+        secrets: dict = None,
+    ):
+        """Delete an artifact object in the DB and optionally delete the artifact data
+
+        :param item: Artifact object (can be any type, such as dataset, model, feature store).
+        :param deletion_strategy: The artifact deletion strategy types.
```
Comment (alonmr) [inline mlrun/projects/project.py:1608]: same

File: mlrun/common/schemas/artifact.py [hunk 3]
```diff
@@ -93,3 +93,18 @@ class Artifact(pydantic.BaseModel):
...
+
+class ArtifactsDeletionStrategies(mlrun.common.types.StrEnum):
+    """Artifacts deletion strategies types."""
+
+    metadata_only = "metadata-only"
+    """Only removes the artifact db record, leaving all related artifact data in-place"""
+
+    data_optional = "data-optional"
+    """Delete the artifact data of the artifact as a best-effort.
+    If artifact db record deletion fails still try to delete the artifact data"""
```
Comment (alonmr) [inline mlrun/common/schemas/artifact.py]: isn't it the other way around?

**Review by alonmr — COMMENTED** [review]
A few comments / questions

File: mlrun/db/httpdb.py (hunk 1 above)
Comment (moranbental) [reply mlrun/db/httpdb.py:1007]: Why? The parameter type exists in the method signature

**Review by alonmr — APPROVED** [review]
Approved but I still think it's better to be explicit in the docstring (remember it also goes to mlrun docs site). See `mlrun.db.httpdb.HTTPRunDB.delete_project` for reference.

**Pull request merged**